### PR TITLE
Feat category api

### PR DESCRIPTION
**Pull request by Sochima-Aristotle**
#### What does this PR achieve?
- Added buttons for category and sub-category fromcreate attendance page
-  Created ui for category and sub-category
-  connected category and sub-category to the API
#### Screenshots:
image

image

image

#### Relevant Jira Tickets

- [Ticket 1](http://some.com)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const RenderDevTool = () => {
   }
   return null;
 };
-// { process.env.NODE_ENV === 'development' ? <ReactQueryDevtools initialIsOpen={false} /> : null }
```
Comment (chineduknight) [inline src/App.tsx:18]: why did you remove this

File: src/pages/Category.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import _ from 'lodash';
+import { orgRequest } from "services";
+import { queryClient, useMutationWrapper } from "services/api/apiHelper";
+
+
+const Category = () => {
+    const onSuccess = (data) => {
+        queryClient.invalidateQueries({ queryKey: ["all-organisations"] })
+    }
+
```
Comment (chineduknight) [inline src/pages/Category.tsx]: why is this still here

File: src/pages/SubCategory.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import useGlobalStore, { currentAttendanceType } from "zStore";
+import _ from 'lodash';
+import { postRequest, queryClient, useMutationWrapper } from "services/api/apiHelper";
+import { orgRequest } from "services";
+
+const SubCategory = () => {
+    const onSuccess = (data) => {
+        queryClient.invalidateQueries({ queryKey: ["all-organisations"] })
+    }
+
```
Comment (chineduknight) [inline src/pages/SubCategory.tsx]: I don't understand why this is here

**Review by chineduknight — CHANGES_REQUESTED** [review]
You have copied and pasted somethings from the create the organisation page,
but you forget to modify all that needs to me modified 
hence this is still pending

File: src/App.tsx (hunk 1 above)
Comment (Sochima-Aristotle) [reply src/App.tsx:18]: I was checking something, that code wasn't neccessay.

File: src/pages/Category.tsx (hunk 2 above)
Comment (Sochima-Aristotle) [reply src/pages/Category.tsx]: I have removed it.

File: src/pages/SubCategory.tsx (hunk 3 above)
Comment (Sochima-Aristotle) [reply src/pages/SubCategory.tsx]: resolved

**Comment by Sochima-Aristotle** [comment]
I have resolved all the issues raised from the last PR, sorry about my late responds, I promise to do better next time both in communication and at work.

thanks

File: src/pages/CreateAttendance.tsx [hunk 4]
```diff
@@ -15,6 +15,7 @@ import { PROTECTED_PATHS } from "routes/pagePath";
 import { useForm, SubmitHandler } from "react-hook-form";
 import useGlobalStore, { currentAttendanceType } from "zStore";
 import _ from 'lodash';
+// import { queryClient, useMutationWrapper } from "services/api/apiHelper";
```
Comment (chineduknight) [inline src/pages/CreateAttendance.tsx:18]: remove this line

File: src/pages/Category.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+import { queryClient, useMutationWrapper } from "services/api/apiHelper";
+import { convertParamsToString } from "helpers/stringManipulations";
+
+
+const Category = () => {
+    const [org] = useGlobalStore((state) => [state.organisation]);
+    const onSuccess = (data) => { }
+    // const modelURL = convertParamsToString(orgRequest.CONFIG_MODEL, {
+    //     organisationId: org.id,
+    // });
```
Comment (chineduknight) [inline src/pages/Category.tsx]: remove this line

File: src/pages/Category.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    Text,
+    Button,
+    FormControl,
+    FormLabel,
+    Input,
+    Stack,
+} from "@chakra-ui/react";
+
+import { useNavigate } from "react-router-dom";
+// import { PROTECTED_PATHS } from "routes/pagePath";
```
Comment (chineduknight) [inline src/pages/Category.tsx]: remove all the comments

**Review by chineduknight — CHANGES_REQUESTED** [review]
Remove commented code

**Review by chineduknight — APPROVED** [review]
looks good
